catalog/transformer/catalog-transformer-csv/src/main/java/ddf/catalog/transformer/csv/CsvQueryResponseTransformer.java

```diff
@@ -91,7 +91,8 @@ public BinaryContent transform(
             .orElse(Collections.emptyMap());
 
     Set<AttributeDescriptor> allAttributeDescriptors =
-        getAllRequestedAttributes(upstreamResponse.getResults(), hiddenFields);
+        getAllRequestedAttributes(
+            upstreamResponse.getResults(), hiddenFields, new HashSet<String>(attributeOrder));
 
```

Comment:
nit - I'd suggest creating a new variable called `requestedFields` instead of creating the `HashSet` in-line.  Something like `Set<String> requestedFields = new HashSet<String>(attributeOrder)`.  The result is the same, but it helps keep the code readable for others.

Reply:
Fixed bcb6500b403ee42e5cc98f26b1ca7839f9430aea